  @Test(expected = ConfigException.class)
  public void throwAnExceptionWhenForANewTableToCreateWhichDoesNotAllowAutoCreation() throws SQLException {
    Database changesExecutor = new Database(new HashSet<String>(),
            new HashSet<String>(),
            new DatabaseMetadata(null, new ArrayList<DbTable>()),
            DbDialect.fromConnectionString(SQL_LITE_URI),
            2);

    String tableName = "tableA";
    Map<String, Collection<SinkRecordField>> map = new HashMap<>();
    map.put(tableName, Lists.newArrayList(
            new SinkRecordField(Schema.Type.INT32, "col1", true),
            new SinkRecordField(Schema.Type.STRING, "col2", false),
            new SinkRecordField(Schema.Type.INT8, "col3", false),
            new SinkRecordField(Schema.Type.INT64, "col3", false),
            new SinkRecordField(Schema.Type.FLOAT64, "col4", false)
    ));

    Connection connection = null;
    try {
      connection = connectionProvider.getConnection();
      changesExecutor.update(map, connection);
    } finally {
      AutoCloseableHelper.close(connection);
    }
  }